### ISSUE TITLE
ERROR: No matching distribution found for onnxruntime==1.7.0

### ISSUE THREAD

**Issue by etetteh** [ivadomed/ivadomed#1057]
I keep getting the following error, when installing `ivadomed` from source on Compute Canada (graham).
```ERROR: No matching distribution found for onnxruntime==1.7.0```

Whether with `python==3.9.7` or `python==3.8`, and
`pip install torch==1.8.1+cu111 torchvision==0.9.1+cu111 --find-links https://download.pytorch.org/whl/torch_stable.html`, I still run into the same error.

**Comment by dyt811** [comment]
So apparently Compute Canada has very limited package availability versus regular PyPI. @naga-karthik ran into this similar issue before. Did you manage to get it resolved, Naga?

**Comment by naga-karthik** [comment]
Yeah, the `onnxruntime` package causes a problem on Compute Canada. @etetteh I was not able to resolve it but one quick way to fix this is to comment out the import statement in [this](https://github.com/ivadomed/ivadomed/blob/master/ivadomed/inference.py#L3) line. This works because onnxruntime is only used when we want to run inference on a system that does not have a GPU, in which we have to convert it into the system-agnostic onnx format so that it also works on the CPU. 

Hence, commenting it out should not cause any issue in routine ivadomed training.

**Comment by etetteh** [comment]
Thanks. I commented `onnxruntime` out both [here](https://github.com/ivadomed/ivadomed/blob/97b5772374660b12895f7a458941f468090e8bf8/ivadomed/inference.py#L3) and [here](https://github.com/ivadomed/ivadomed/blob/97b5772374660b12895f7a458941f468090e8bf8/requirements_common.txt#L13), and that issue was resolved.

However, installing ivadomed on Compute Canada inside a `python==3.9` virtual environment raised the following error:
![Screen Shot 2022-01-31 at 13 33 06](https://user-images.githubusercontent.com/28512232/151869972-a421a716-f5f9-4b85-87f6-936255694de0.JPG)

The fix for a successful installation for me was to use `python==3.7.9`.

**Comment by dyt811** [comment]
Thanks for reporting back! Here s another related issue #919.

**Comment by etetteh** [comment]
Thank you @dyt811. As mentioned by @naga-karthik onnxruntime is not needed when running on GPU platform, so simply commenting it out in the requirements_common.txt file and the import in ivadomed/inference.py was the fix.

**Comment by kanishk16** [comment]
@dyt811 #1000 would install onnxruntime only when user explicitly specifies (not added as of now, but definitely had this in mind while drafting the PR). Perhaps we could consider importing onnxruntime inside https://github.com/ivadomed/ivadomed/blob/422ada13a3d4d4904116ab99dac7d2cb61ac841a/ivadomed/inference.py#L31-L45

which would basically import onnxruntime only when the onnx_inference is called. I guess this should resolve this issue. Moreover, it's worth checking if this is due to the strict requirements as Joshua recommended loosening our requirements a bit.

**Comment by jcohenadad** [comment]
@mpompolas experienced the same issue on the servers at McGill. I think it is reasonable to increase the priority level of this issue

**Comment by kousu** [comment]
This also shows up on any system [using python3.10](https://github.com/ivadomed/ivadomed/issues/1136):

https://github.com/ivadomed/ivadomed/runs/6658529084?check_suite_focus=true#step:4:68

```
ERROR: Ignored the following versions that require a different python version: 2.5.2 Requires-Python !=3.0.*,!=3.1.*,!=3.2.*,!=3.3.*,<3.9dev,>=2.7
ERROR: Could not find a version that satisfies the requirement onnxruntime==1.7.0 (from versions: none)
ERROR: No matching distribution found for onnxruntime==1.7.0
Error: ERROR: Action failed during dependency installation attempt with error: The process '/opt/hostedtoolcache/Python/3.10.4/x64/bin/python' failed with exit code 1
```

Unfortunately onnx doesn't yet support 3.10: https://github.com/microsoft/onnxruntime/issues/11680
